plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/DraggableAnnotationController.java

```diff
@@ -16,17 +16,32 @@
 import androidx.annotation.Nullable;
 import androidx.annotation.VisibleForTesting;
 
-final class DraggableAnnotationController<T extends Annotation, D extends OnAnnotationDragListener<T>> {
+import java.util.ArrayList;
+import java.util.List;
+
+final class DraggableAnnotationController <T extends Annotation, D extends OnAnnotationDragListener<T>> {
+
+  private static DraggableAnnotationController INSTANCE = null;
+
+  public static synchronized DraggableAnnotationController getInstance(MapView mapView, MapboxMap mapboxMap) {
+    if(INSTANCE == null){
+      INSTANCE = new DraggableAnnotationController(mapView, mapboxMap);
+    }
+    return INSTANCE;
+  }
```

Comment:
An issue with this singleton approach might occur when the map is recreated, for example in a recycler view, or when transitioning between fragments/activites.

We might need to do something like this, to additionally ensure that we're using the correct instance:
```suggestion
  public static synchronized DraggableAnnotationController getInstance(MapView mapView, MapboxMap mapboxMap) {
    if(INSTANCE == null || this.mapView != mapview || this.mapboxMap != mapboxMap){
      INSTANCE = new DraggableAnnotationController(mapView, mapboxMap);
    }
    return INSTANCE;
  }
```

The strong references to `mapView` and `mapboxMap` should be dropped when the last `removeAnnotationManager` is called and the instance is nullified so we shouldn't leak. It'd be great to test this scenario though.

Reply:
You're right I did not think about that 👍. I updated the code to reflect what you said. Regarding the testing of the last part of your remark, I'm not sure how I can do it.